web/components/modals/CaretakerModal.tsx

```diff
@@ -0,0 +1,76 @@
+import React from "react";
+import { useForm } from "react-hook-form";
+import { BiLoaderCircle } from "react-icons/bi";
+import { FaTimes } from "react-icons/fa";
+
+function CaretakerModal({
+  setModal,
+  onSubmit,
+  buttonText = "Add caretaker",
+  header = "Add a Caregiver",
+  values = {},
+}: {
+  setModal: (modal: boolean) => void;
+  onSubmit: (data: any) => void;
+  buttonText?: string;
+  header?: string;
+  values?: any;
+}) {
+  const { register, handleSubmit, formState: { errors, isSubmitting } } = useForm();
+
+  const handleClose = () => setModal(false);
+
+  return (
```

Comment:
Would you be able to add the veil around the modal (grey page dim) that closes the modal when the user clicks on it? This is another thing where we can add later if you don't have time this sprint just lmk
![image](https://github.com/user-attachments/assets/918e3c20-6e35-4bb4-b008-40af1c4f1afe)